src/dataframe/dataframe.jl

```diff
@@ -122,9 +123,17 @@ end
 
 function DataFrame(columns::AbstractVector,
                    cnames::AbstractVector{Symbol} = gennames(length(columns)))
+    if !all(col -> isa(col, AbstractVector), columns)
+        # change to throw(ArgumentError("columns argument mutst be a vector of vectors"))
+        Base.depwarn("passing columns argument other than a vector of vectors is deprecated", :DataFrame)
+    end
     return DataFrame(convert(Vector{Any}, columns), Index(convert(Vector{Symbol}, cnames)))
 end
 
+# if #1308 is merged this sould be refactor to include makeunique
```

Comment:
Let's merge #1308 first.

Reply:
OK. I am pushing the as-is version to have it on GitHub, but I will rebase it when 1308 is merged.